### DISCUSSION TITLE
including rpc calls in bash script

### DISCUSSION THREAD

**Discussion by iFriendGit** [jamulussoftware/jamulus#2762] [Technical Support & Help] [answered]
Hello everyone, please how do you go about correctly including rpc calls to Jamulus into a bash script? 

What I tried is:
_#!/bin/bash
nc localhost 7938 < rpc_msgs.txt
other cmds.._

Where rpc_msgs.txt is:
{"id":1,"jsonrpc":"2.0","method":"jamulus/apiAuth","params":{"secret": "secret"}}
{"id":1,"jsonrpc":"2.0","method":"jamulusclient/setName","params":{"name": "Listener"}}

The problem though is that netcat does not exit, so the script is stuck.

My workaround atm is to add some garbage msg, so that netcat quits:
_{"id":1,"jsonrpc":"2.0","method":"jamulus/apiAuth","params":{"secret": "secret"}}
{"id":1,"jsonrpc":"2.0","method":"jamulusclient/setName","params":{"name": "Listener"}}
garbageMsg_

But that's not the proper way I guess. So do you guys have any hint, please?
Thank you in advance!

**Comment by pljones** [comment, marked as answer]
Maybe add `-N`?
```
     -N      shutdown(2) the network socket after EOF on the input.  Some servers require this to finish their work.
```

**Reply by iFriendGit** [reply]
thank you pljones! That's it!
